lib/mosaico/active_storage_backend.rb

```diff
@@ -0,0 +1,31 @@
+module Mosaico
+  class ActiveStorageBackend
+    attr_reader :service, :path_prefix
+
+    def initialize(service, path_prefix)
+      @service = service
+      @path_prefix = path_prefix
+    end
+
+    def store(source, as:)
+      key = "#{path_prefix}/#{as}"
+      @service.upload(key, open(source))
+    end
+
+    def retrieve(file)
+      key = "#{path_prefix}/#{file}"
+      data = @service.download(key)
+
+      tmp_file = Tempfile.new(file, :encoding => 'ascii-8bit')
+      tmp_file.write(data)
+      tmp_file.close
+
+      tmp_file.path
+    end
+
+    def url_to(file)
+      key = "#{path_prefix}/#{file}"
+      @service.url(key, filename: ActiveStorage::Filename.new(file), expires_in: 1.week, disposition: :inline, content_type: nil)
```

Comment:
I'm not very familiar with active storage, but do we really want the images to expire?

Reply:
I also wanted to avoid that if possible, but it's a required param for certain cloud storage services. This won't actually expire the image though. It just returns an short-lived service_url of the image which will only be valid before it expires. It will return a new service_url when the current one, if exists, is about to expire. Since mosaico image controller redirects to this url, I should be OK for it to be short lived in most cases. I'd say it's unnecessary, but doesn't really hurt either.

Once I get a chance I may want to change it to blob_url, which do not expire. However the doc on ActiveStorage::Blob is not very detailed. So I probably need to spend some time digging into the ActiveStorage source code.